File: PlantEngine/app/src/main/java/com/example/plantengine/MainActivity.java
package com.example.plantengine;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {
    ImageButton WaterBtn, SettingBtn, PumpBtn, LedBtn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        WaterBtn = (ImageButton)findViewById(R.id.waterbutton);
        SettingBtn = (ImageButton) findViewById(R.id.setting);
        PumpBtn = (ImageButton)findViewById(R.id.pumpbutton);
        LedBtn = (ImageButton) findViewById(R.id.ledbutton);

        WaterBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), WaterTankActivity.class);
                startActivity(intent);
            }
        });

        SettingBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent2 = new Intent(getApplicationContext(), SettingActivity.class);
                startActivity(intent2);
            }
        });

        PumpBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent3 = new Intent(getApplicationContext(), PumpActivity.class);
                startActivity(intent3);
            }
        });

        LedBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent4 = new Intent(getApplicationContext(), LEDActivity.class);
                startActivity(intent4);
            }
        });
    }
}

File: PlantEngine_TermProject/app/src/main/java/com/example/plantengine_termproject/MainActivity.kt
package com.example.plantengine_termproject

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ListView
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        ClickLister()
    }


    private fun ClickLister(){
        waterbutton.setOnClickListener {
            val intent: Intent = Intent(this, WaterTankActivity::class.java)
            startActivity(intent)
        }

        setting.setOnClickListener {
            val intent:Intent = Intent(this, SettingActivity::class.java)
            startActivity(intent)
        }
        pumpbutton.setOnClickListener {
            val intent:Intent = Intent(this, PumpActivity::class.java)
            startActivity(intent)
        }
        ledbutton.setOnClickListener {
            val intent:Intent = Intent(this, LEDActivity::class.java)
            startActivity(intent)
        }
    }
}